tests/test_sbt.py

```diff
@@ -470,21 +470,22 @@ def test_save_zip(tmpdir):
 
     shutil.copyfile(testdata, str(testsbt))
 
+    to_search = load_one_signature(utils.get_test_data(utils.SIG_FILES[0]))
+    search_obj = make_jaccard_search_query(threshold=0.1)
+
     tree = SBT.load(str(testsbt), leaf_loader=SigLeaf.load)
+    old_result = {str(s.signature) for s in tree.find(search_obj, to_search)}
     tree.save(str(newsbt))
+
     assert newsbt.exists()
 
     new_tree = SBT.load(str(newsbt), leaf_loader=SigLeaf.load)
     assert isinstance(new_tree.storage, ZipStorage)
     assert new_tree.storage.list_sbts() == ['new.sbt.json']
-
-    to_search = load_one_signature(utils.get_test_data(utils.SIG_FILES[0]))
+    new_result = {str(s.signature) for s in new_tree.find(search_obj, to_search)}
 
     print("*" * 60)
     print("{}:".format(to_search))
-    search_obj = make_jaccard_search_query(threshold=0.1)
-    old_result = {str(s.signature) for s in tree.find(search_obj, to_search)}
-    new_result = {str(s.signature) for s in new_tree.find(search_obj, to_search)}
```

Comment:
OK, this is fun: I reordered the test because the call to `tree.save(str(newsbt))` closes the original zipfile used as storage in `testsbt`. Why did it trigger only when defaulting to `unload_data=True`? Because with `unload_data=False` the data was still in memory, even with the storage zipfile closed.

the call to `.close()` inside `.save()` should probably be something like `.flush()` instead, which might fix the warnings about closed files too...